### DISCUSSION TITLE
Allow reviewers to comment on commit messages

### DISCUSSION THREAD

**Discussion by networkException** [community/community#4983] [Pull Requests]
Keeping a clean git history that gives background on motivation and design choices for specific changes is extremely valuable to a project. Currently when reviewing a pull request, it is not possible to comment on or suggest changes to commit messages.

It would be great if reviewing commit messages could be added!

**Comment by xmo-odoo** [comment]
Also quite simply when reading commit messages sometimes typos or unfortunate wording is spotted (especially when english is not the mother tongue of many or most team members).

Worse, the smaller the error -- and the more detailed the message -- the more difficult and frustrating it is to notify the author of issues: if the entire thing should be rewritten it's easy to just provide a proposal as a quote.

**Reply by MasterInQuestion** [reply]
I think the ability to refactor commits is the key to solve alike problems:
    https://github.com/orgs/community/discussions/56449#discussioncomment-10815660

**Comment by jbrandeb** [comment]
Especially for people like myself who have to follow a 3rd party (Linux kernel - you may have heard of it) guidelines for submitting patches, being able to comment on commit messages as part of PR review is *really* important.

**Comment by ADKaster** [comment]
This feature is critical for projects that follow a rebase workflow, rather than a squash one. When we pull commits from a PR in SerenityOS/serenity directly on top of master, we have a GitHub Action that lints the commit message and posts a bot response if there's a problem with the message. However, even if the commit passes the style check, there could still be grammar, spelling, tone, or other issues with the commit message itself that are difficult to comment on without commenting on the PR body with things like "The commit message `LibFoo: Adds the new feature` has the wrong style, it should be something like `LibFoo: Use feature to do Y`".

**Comment by DevSDK** [comment]
Gerrit user here. I think we really need is adding a review system (including comments on the commit message) to the message for the squash merge commit.

For example [1], The reviewer checked the too-long commit message and I could change it **BEFORE** merging to the main branch.

The github for now that is always **on the author**. We cannot check early because the squash-merge commit is written by textarea after the reviewer-approves.

This can easily cause the consistent quality of commit messages to break.

If GitHub had something similar, it would make sure the commit messages are always up to scratch, which is great for overall code quality.

[1] https://chromium-review.googlesource.com/c/chromium/src/+/3713201/14..17//COMMIT_MSG#b13

**Comment by DevSDK** [comment]
Friendly ping

**Comment by DevSDK** [comment]
Again, friendly ping
